core/src/main/scala/org/apache/spark/serializer/SerializationDebugger.scala

```diff
@@ -110,8 +112,13 @@ private[spark] object SerializationDebugger extends Logging {
             val elem = s"externalizable object (class ${e.getClass.getName}, $e)"
             visitExternalizable(e, elem :: stack)
 
-          case s: Object with java.io.Serializable =>
-            val elem = s"object (class ${s.getClass.getName}, $s)"
+          case s: Object with java.io.Serializable if Utils.isTesting =>
```

Comment:
Dont we need this when not in testing ?

Reply:
hmm... currently, no similar cases have been created in non-testing scenarios, right? @summaryzb

Reply:
All sql  scenarios has the same semantic in non-testing ,  but in other cases that the  `Serializable` object call `toString` will not throw still need this. We should change it like below to keep the  same semantic  as previous
```
          case s: Object with java.io.Serializable =>
            val str = try {
              s.toString
            } catch {
              case _: SparkRuntimeException if Utils.isTesting => "exception in toString"
              case e: Throwable => throw e
            }
```

Reply:
@the-sakthi I submit a [followup](https://github.com/apache/spark/pull/50489) to solving this problem to make SerializationDebugger still work in non-testing scenario

Reply:
> Dont we need this when not in testing ?

I'm very sorry, this was an incorrect merge due to my misreading of the code. Let me revert it first. @summaryzb @mridulm